components/PriceFeedTable.tsx

```diff
@@ -0,0 +1,119 @@
+      ? priceFeeds
+      : priceFeeds.filter((feed) => feed.assetType === selectedAssetType);
+
+  const copyToClipboard = useCallback((text: string) => {
+    navigator.clipboard
+      .writeText(text)
+      .then(() => {
+        setCopiedId(text);
+        setTimeout(() => setCopiedId(null), 2000); // Hide the popup after 2 seconds
+      })
+      .catch((err) => {
+        console.error("Failed to copy: ", err);
+      });
+  }, []);
+
+  return (
+    <div>
+      <div className="mb-4">
+        <label htmlFor="assetTypeFilter" className="mr-2">
+          Filter by Asset Type:
+        </label>
+        <select
+          id="assetTypeFilter"
+          value={selectedAssetType}
+          onChange={(e) => setSelectedAssetType(e.target.value)}
+          className="p-2 border rounded"
+        >
+          {assetTypes.map((type) => (
+            <option key={type} value={type}>
+              {type}
+            </option>
+          ))}
+        </select>
+      </div>
+      <table>
+        <thead>
+          <tr>
+            <th className={columnWidths.assetType}>Asset Type</th>
+            <th className={columnWidths.name}>Name</th>
+            <th className={columnWidths.ids}>Feed ID</th>
+          </tr>
+        </thead>
+        <tbody>
+          {filteredFeeds.map((feed, index) => (
+            <tr key={index}>
+              <StyledTd>{feed.assetType}</StyledTd>
+              <StyledTd>{feed.name}</StyledTd>
+              <StyledTd>
+                <div className="relative">
+                  <button
+                    onClick={() => copyToClipboard(feed.ids)}
+                    className="flex items-center space-x-2 px-2 py-1 bg-gray-100 hover:bg-gray-200 dark:bg-darkGray2 dark:hover:bg-darkGray3 rounded transition duration-200"
+                  >
+                    <code className="dark:text-darkLinks text-lightLinks dark:bg-darkGray2 bg-gray-100 px-2 py-1 rounded">
+                      {feed.ids}
+                    </code>
+                    <svg
+                      xmlns="http://www.w3.org/2000/svg"
+                      className="h-4 w-4"
+                      fill="none"
+                      viewBox="0 0 24 24"
+                      stroke="currentColor"
+                    >
+                      <path
+                        strokeLinecap="round"
+                        strokeLinejoin="round"
+                        strokeWidth={2}
+                        d="M8 16H6a2 2 0 01-2-2V6a2 2 0 012-2h8a2 2 0 012 2v2m-6 12h8a2 2 0 002-2v-8a2 2 0 00-2-2h-8a2 2 0 00-2 2v8a2 2 0 002 2z"
+                      />
+                    </svg>
+                  </button>
+                  {copiedId === feed.ids && (
+                    <div className="absolute bottom-full left-1/2 transform -translate-x-1/2 mb-2 px-2 py-1 bg-gray-800 text-white text-sm rounded shadow">
+                      Copied to clipboard
+                    </div>
+                  )}
+                </div>
```

Comment:
A lot of this component is doing the same stuff that's already in Nextra in [`copy-to-clipboard.tsx`](https://github.com/shuding/nextra/blob/main/packages/nextra/src/client/components/copy-to-clipboard.tsx), [`code.tsx`](https://github.com/shuding/nextra/blob/main/packages/nextra/src/client/components/code.tsx), and [`pre.tsx`](https://github.com/shuding/nextra/blob/5834194fdde2aace505298fa74603aadb77b0b10/packages/nextra/src/client/components/pre.tsx#L6).  There's also built-in table components too.  I'd advise trying to find a way to leverage the nextra built in components for this stuff -- e.g. I'm guessing you can replace this entire component with something very similar to this:

```tsx
import { Code, Pre, Table, Td, Th, Tr } from 'nextra/components';

const PriceFeedTable = ({ priceFeeds }: { priceFeeds: PriceFeed[] }) => {
  // .... filtering stuff

  return (
    <Table>
      <thead>
        <Tr>
          <Th className="w-3/10">Asset Type</Th>
          <Th className="w-1/5">Name</Th>
          <Th className="w-1/2">Feed ID</Th>
        </Tr>
      </thead>
      <tbody>
        {filteredFeeds.map(({ assetType, name, ids }) => ({
          <Tr key={name}>
            <Td>{assetType}</Td>
            <Td>{name}</Td>
            <Td>
              <Pre data-copy="">
                <Code>
                  {name}
                </Code>
              </Pre>
            </Td>
          </Tr>
          }))}
      </thead>
    </Table>
  );
};
```

Note `pre` is short for Preformatted and is nearly always the right html tag to wrap a code block, see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/pre.

Not only will this drastically reduce the amount of custom code here, but you'll also make things much more consistent with the rest of the site.